src/services/auditLogService.ts

```diff
@@ -0,0 +1,43 @@
+import * as gqlTypes from '../typeDefs/gqlTypes.js'
+import * as dbSchema from '../typeDefs/dbSchema.js'
+// import { ErrorCode, Result } from '../result.js'
+import { dbInstance } from '../firebaseDb.js'
+import { logger } from '../logger.js'
+
+type CreatedAuditLog = {
+    isSuccesful: boolean
+}
+
+export async function createAuditLog(auditLogInput: dbSchema.AuditLog): Promise<CreatedAuditLog> {
+    try {
+        // TO DO: Validations here
+
+        const auditLogRef = dbInstance.collection('auditLogs').doc()
+        const newAuditLogId = auditLogRef.id
+        const newAuditLog = mapGqlEntityToDbEntity(auditLogInput, newAuditLogId)
+
+        await auditLogRef.set(newAuditLog)
```

Comment:
I assume when the set fails it would catch it in the catch block (to much GO the last week)

Reply:
haha you got it! you can actually create more than one catch for specific errors. So in this example, if you know the error types that might be thrown from the set, you can handle them differently than the generic error handler. 
In this case, I'm not sure how much we would do differently other than something like a retry 🤔